plugins/elantp/fu-elantp-i2c-device.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include <linux/i2c-dev.h>
+
+#include "config.h"
+
+#include "fu-elantp-common.h"
+#include "fu-elantp-i2c-device.h"
+#include "fu-chunk.h"
+
+struct _FuElantpI2cDevice {
+	FuUdevDevice		 parent_instance;
+	guint16			 iap_ctrl;
+	guint16			 ic_page_count;
+};
+
+G_DEFINE_TYPE (FuElantpI2cDevice, fu_elantp_ic2_device, FU_TYPE_UDEV_DEVICE)
+
+static gboolean
+fu_elantp_i2c_device_send_cmd (FuElantpI2cDevice *self,
+			       guint8 *tx, gssize txsz,
+			       guint8 *rx, gssize rxsz,
+			       GError **error)
+{
+	if (!fu_udev_device_pwrite_full (FU_UDEV_DEVICE (self), 0, tx, txsz, error))
+		return FALSE;
+	if (rxsz == 0)
+		return TRUE;
+	if (!fu_udev_device_pread_full (FU_UDEV_DEVICE (self), 0, rx, rxsz, error))
+		return FALSE;
+	return TRUE;
+}
+
+G_GNUC_UNUSED static gboolean
+fu_elantp_i2c_device_write_cmd (FuElantpI2cDevice *self, guint16 reg, guint16 cmd, GError **error)
+{
+	guint8 buf[4];
+	fu_common_write_uint16 (buf + 0x0, reg, G_LITTLE_ENDIAN);
+	fu_common_write_uint16 (buf + 0x2, cmd, G_LITTLE_ENDIAN);
+	if (g_getenv ("FWUPD_ELANTP_VERBOSE") != NULL)
+		fu_common_dump_raw (G_LOG_DOMAIN, "WriteCmd", buf, sizeof(buf));
+	return fu_elantp_i2c_device_send_cmd (self, buf, sizeof(buf), NULL, 0, error);
+}
+
+G_GNUC_UNUSED static gboolean
+fu_elantp_i2c_device_read_cmd (FuElantpI2cDevice *self, guint16 reg,
+			       guint8 *rx, gsize rxsz, GError **error)
+{
+	guint8 buf[2];
+	fu_common_write_uint16 (buf + 0x0, reg, G_LITTLE_ENDIAN);
+	if (g_getenv ("FWUPD_ELANTP_VERBOSE") != NULL)
+		fu_common_dump_raw (G_LOG_DOMAIN, "ReadCmd", buf, sizeof(buf));
+	return fu_elantp_i2c_device_send_cmd (self, buf, sizeof(buf),
+					      rx, rxsz, error);
+}
+
+static gboolean
+fu_elantp_i2c_device_ensure_iap_ctrl (FuElantpI2cDevice *self, GError **error)
+{
+	return TRUE;
```

Comment:
missing code?  Or should this function not be here?

Reply:
Ya, I need to copy the logic from the HID version. It's more of a placeholder for now.

Reply:
If it's a copy of the logic that's also asking to just split out to a common function too.

Reply:
Will do, but not by me :)